### PR TITLE
Update plugin contribution docs

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Update plugin contribution docs

#### Ticket Link
https://docs.google.com/spreadsheets/d/1X6FXpQFLSXPlLSQOYzEXqWEpgZ_z3hWLhHNcqMr-oxM

### CODE REVIEW

File: site/content/contribute/server/plugins.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Plugins are generally made of at least two parts: a manifest and a server binary
...
+The server binary is a compiled go program that extends the [MattermostPlugin](https://godoc.org/github.com/mattermost/mattermost-server/v6/plugin#MattermostPlugin) struct of the [plugin](https://godoc.org/github.com/mattermost/mattermost-server/v6/plugin) package. When enabled, the plugin's server binary is started as a process by the Mattermost server. Plugin builders then have access to interact with the Mattermost server over RPC through the plugin [API](https://godoc.org/github.com/mattermost/mattermost-server/v6/plugin#API) and [Hooks](https://godoc.org/github.com/mattermost/mattermost-server/v6/plugin#Hooks). The server-side of plugins is built using the [go-plugin](https://github.com/hashicorp/go-plugin) library from Hashicorp. More information is available in the [server side of the plugin author documentation]({{< ref "/integrate/plugins/server" >}}).
```
Comment (mickmister) [inline site/content/contribute/server/plugins.md:23]: Should the language's name really be lowercased? JavaScript is capitalized below

Comment (mickmister) [reply site/content/contribute/server/plugins.md:23]:
```suggestion
The server binary is a compiled go program that extends the [MattermostPlugin](https://godoc.org/github.com/mattermost/mattermost-server/v6/plugin#MattermostPlugin) struct of the [plugin](https://godoc.org/github.com/mattermost/mattermost-server/v6/plugin) package. When enabled, the plugin's server binary is started as a process by the Mattermost server. Plugin creators (or authors?) then have access to interact with the Mattermost server over RPC through the plugin [API](https://godoc.org/github.com/mattermost/mattermost-server/v6/plugin#API) and [Hooks](https://godoc.org/github.com/mattermost/mattermost-server/v6/plugin#Hooks). The server-side of plugins is built using the [go-plugin](https://github.com/hashicorp/go-plugin) library from Hashicorp. More information is available in the [server side of the plugin author documentation]({{< ref "/integrate/plugins/server" >}}).
```

https://developers.mattermost.com/integrate/plugins/server/reference seems more digestible as a reference for the RPC API. https://developers.mattermost.com/integrate/plugins/server/reference/#Hooks for hooks. But this is for contributing to the framework itself, so maybe not relevant.

Also, we never mention `mattermost-plugin-api` in this page nor the server reference page. There is a `Helpers` section there that is blank.

**Review by mickmister — CHANGES_REQUESTED** [review]
Great work @hanzei :+1: I really like this documentation iniitiative here.

> In the future there will be another component of plugins for the mobile apps, likely a React Native bundle.

We probably want to remove this or mention Apps instead

> Once it is enabled, the server will then start the server process and prepare the web app bundle for serving to the client.

The server will notify the webapp to fetch the bundle as well

> and [mattermost-plugin-demo](https://github.com/mattermost/mattermost-plugin-demo) for an example plugin.

Do we want to also mention the starter template here?

**Review by cwarnermm — APPROVED** [review]
Great update! Thank you!

**Pull request merged**